### PR TITLE
server: return mvcc key and value for mvcc http api

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
1. fix https://github.com/pingcap/tidb/issues/9915, out put the key of mvcc value.
Old TiDB: 

```shell
▶ curl "http://$IP:10080/mvcc/index/test/t/a/1?a=1"
{
 "info": {
  "writes": [
   {
    "start_ts": 407308196035690497,
    "commit_ts": 407308196035690500,
    "short_value": "AAAAAAAAAAE="
   }
  ]
 }
}%                                                                                                                                                                                      
github.com/pingcap/tidb  master ✔                                                                                                                                                  1d
▶ curl "http://$IP:10080/mvcc/key/test/t/1"
{
 "info": {
  "writes": [
   {
    "start_ts": 407308196035690497,
    "commit_ts": 407308196035690500,
    "short_value": "CAIIAggEAhjlk4jlk4ggaGVsbG8IBgAICAmAgIDIxo6u0Rk="
   }
  ]
 }
}% 
```

Now in this PR:
```shell
▶ curl "http://$IP:10080/mvcc/index/test/t/a/1?a=1"
{
 "key": "7480000000000000495F698000000000000001038000000000000001",
 "value": {
  "info": {
   "writes": [
    {
     "start_ts": 407308196035690497,
     "commit_ts": 407308196035690500,
     "short_value": "AAAAAAAAAAE="
    }
   ]
  }
 }
}% 
                                                                                                                                                                              
github.com/pingcap/tidb  http-mvcc-key ✔                                                                                                                                           1m
▶ curl "http://$IP:10080/mvcc/key/test/t/1"
{
 "key": "7480000000000000495F728000000000000001",
 "value": {
  "info": {
   "writes": [
    {
     "start_ts": 407308196035690497,
     "commit_ts": 407308196035690500,
     "short_value": "CAIIAggEAhjlk4jlk4ggaGVsbG8IBgAICAmAgIDIxo6u0Rk="
    }
   ]
  }
 }
}%                                                                                                                                                                                  
```

2. add deocode paramater. @lysu 
```shell
▶ curl "http://$IP:10080/mvcc/key/test/t/1?decode=1"
{
 "data": {
  "407308196035690497": [
   {
    "a": "1"
   },
   {
    "b": "哈哈 hello"
   },
   {
    "c": "nil"
   },
   {
    "d": "2019-03-28 07:17:41"
   }
  ]
 },
 "info": {
  "writes": [
   {
    "start_ts": 407308196035690497,
    "commit_ts": 407308196035690500,
    "short_value": "CAIIAggEAhjlk4jlk4ggaGVsbG8IBgAICAmAgIDIxo6u0Rk="
   }
  ]
 },
 "key": "7480000000000000495F728000000000000001"
}% 
```
### What is changed and how it works?
* Return key value for mvcc http api. And the `Key` format is hex.
* Add `decode` paramater for `http://$IP:10080/mvcc/key/db_name/table_name/` api to decode raw data value with table schema.

### Check List 

Tests 

Code changes

Side effects

 - Breaking backward compatibility

Related changes

### CODE REVIEW

**Comment by crazycs520** [comment]
@breeswish  PTAL

**Comment by lysu** [comment]
@crazycs520  could you convenient to pick this commit by the way 
 https://github.com/lysu/tidb/commit/eca9aef2a0aa57f0ea2658e90bb56aa1dd5b1f66 thx :D

**Comment by crazycs520** [comment]
@lysu , Done, PTAL.

**Comment by breeswish** [comment]
Should we adopt https://github.com/tikv/rfcs/blob/master/text/2018-11-12-unified-key-format.md for keys?

**Comment by zz-jason** [comment]
> Should we adopt [tikv/rfcs:text/2018-11-12-unified-key-format.md@`master`](https://github.com/tikv/rfcs/blob/master/text/2018-11-12-unified-key-format.md) for keys?

@breeswish What key format is `tikv-ctl` using now?

File: server/http_handler.go [hunk 1]
```diff
@@ -161,9 +161,15 @@ func (t *tikvHandlerTool) getMvccByEncodedKey(encodedKey kv.Key) (*kvrpcpb.MvccG
 	return kvResp.MvccGetByKey, nil
 }
 
-func (t *tikvHandlerTool) getMvccByHandle(tableID, handle int64) (*kvrpcpb.MvccGetByKeyResponse, error) {
+type mvccKV struct {
+	Key   string      `json:"key"`
+	Value interface{} `json:"value"`
```
Comment (zz-jason) [inline server/http_handler.go]: can we use `*kvrpcpb.MvccGetByKeyResponse` as the value type?

**Comment by breeswish** [comment]
@zz-jason It accepts hex. But.. it accepts raw key instead of TiDB key TAT

File: server/http_handler.go (hunk 1 above)
Comment (crazycs520) [reply server/http_handler.go]: done.

**Comment by crazycs520** [comment]
@breeswish @zz-jason PTAL, 
currently in this PR,  the key format is hex, the value is rawKey. And `tidb-ctl base64decode `  use rawKey to decode table value.

**Comment by winkyao** [comment]
@breeswish PTAL

**Comment by breeswish** [comment]
LGTM

**Pull request merged**